src/WebApi/Infrastructure/Repositories/QuestionListRepository.cs

```diff
@@ -129,4 +129,24 @@ public async Task<bool> RemoveQuestionsFromList(int questionListId, IEnumerable<
             return false;
         }
     }
+
+    public async Task<bool> UpdateQuestionOrder(int questionSetId, IList<int> orderedQuestionIds)

```

Comment:
Hey @TomasNiemczyk98, I propose we change the data type from `IList` to `IEnumerable`. It's a good practice to use the most general interface possible and I think it should work the same way since the first thing we do here is `orderedQuestionIds.ToArray();`.
But maybe I am missing something 🤔

Reply:
@PlesnikJakub The problem with using an `IEnumerable` here is that it makes no guarantees about the order of the elements, so while an implementation of a `List` will always be enumerated in the provided order, a `HashSet`, for example, might not. Since we are not creating this instance ourselves we could assume that the provided implementation from a controller is a `List`, but this seemed to me like an extra safety measure. If you want to I can of course change it to `IEnumerable`, what do you think? 😃

Reply:
Would ICollection help in this matter? 🤔  if not, feel free to continue with IList :)

Reply:
Same problem as with IEnumerable, so I will leave IList as it is right now 🙂